src/utils/getUniqueIdGenerator.js

```diff
@@ -0,0 +1,8 @@
+function getUniqueIdGenerator() {
+  let index = 0;
+  return () => {
+    return `mui-id-${index++}`;
```

Comment:
@oliviertassinari This id generator is created per context, i.e. per request. If the client and server call the getUniqueId function in the same order then this method is deterministic. And since every request will have it's own context no two requests will have reference to the same `index` variable captured in that function.